### DISCUSSION TITLE
How to release a lock by another process than the owner ?

### DISCUSSION THREAD

**Discussion by maidmaid** [symfony/symfony#57402] [Q&A]
Below, you can see 2 requests. The first one that acquires the lock and the second that tries to release it. Can we do it ?

## Request 1 at 12:00:00

```php
$lock = $this->lockFactory->createLock(resource: 'test', ttl: 60, autoRelease: false);
$lock->acquire();
```

## Request 2 at 12:00:10

```php
$lock = $this->lockFactory->createLock(resource: 'test', ttl: 60, autoRelease: false);
$lock->release(); // do not release
$lock->acquire(); // do not acquire (info log: Failed to acquire the "..." lock. Someone else already acquired the lock.)
```

**Comment by maidmaid** [comment]
Hello @jderusse, could you help me please ? Thanks !

**Comment by ghost** [comment]
Can you show the use case? Why first request does not release the lock?

**Comment by ghost** [comment]
I think in the second request you have to call PersistingStoreInterface instance and delete the Key instance.

**Comment by maidmaid** [comment]
I'm working on a serverless worker that is triggered by the request itself if some messages have been dispatched. When there is no message to handle, the worker is off and costs 0. When some messages have to be handled, the requests triggers the worker. When a worker is running, I'd like to immediately stop the others workers, in the purpose to have only one running worker. To achieve it, I tried to use lock component : the request triggering the worker acquires the lock and once the worker started and finished, it release the lock.
